clients/admin-ui/src/features/system/history/helpers.tsx

```diff
@@ -0,0 +1,247 @@
+import _ from "lodash";
+import React from "react";
+
+import { DictOption } from "~/features/plus/plus.slice";
+import { PrivacyDeclaration } from "~/types/api/models/PrivacyDeclaration";
+import { SystemHistoryResponse } from "~/types/api/models/SystemHistoryResponse";
+import { SystemResponse } from "~/types/api/models/SystemResponse";
+
+// Helper function to format date and time
+export const formatDateAndTime = (dateString: string) => {
+  const date = new Date(dateString);
+  const userLocale = navigator.language;
+
+  const timeOptions: Intl.DateTimeFormatOptions = {
+    hour: "2-digit",
+    minute: "2-digit",
+    hour12: true,
+    timeZoneName: "short",
+  };
+
+  const dateOptions: Intl.DateTimeFormatOptions = {
+    year: "numeric",
+    month: "long",
+    day: "numeric",
+  };
+
+  const formattedTime = date.toLocaleTimeString(userLocale, timeOptions);
+  const formattedDate = date.toLocaleDateString(userLocale, dateOptions);
+
+  return { formattedTime, formattedDate };
+};
+
+export function alignArrays(
+  before: PrivacyDeclaration[],
+  after: PrivacyDeclaration[]
+) {
+  const allNames = new Set([...before, ...after].map((item) => item.data_use));
+  const alignedBefore: PrivacyDeclaration[] = [];
+  const alignedAfter: PrivacyDeclaration[] = [];
+
+  allNames.forEach((data_use) => {
+    const firstItem = before.find((item) => item.data_use === data_use) || {
+      data_use: "",
+      data_categories: [],
+    };
+    const secondItem = after.find((item) => item.data_use === data_use) || {
+      data_use: "",
+      data_categories: [],
+    };
+    alignedBefore.push(firstItem);
+    alignedAfter.push(secondItem);
+  });
+
+  return [alignedBefore, alignedAfter];
+}
+
+const lookupVendorLabel = (vendor_id: string, options: DictOption[]) =>
+  options.find((option) => option.value === vendor_id)?.label ?? vendor_id;
+
+export const getUiLabel = (key: string): string => {
```

Comment:
Phase one of this feature just displayed the backend field names in the UI. This ticket asked us to change just three fields but we might want to change more in the future. When we do, we can just add the mapping to this function.